Parse/SMTLIB2.cpp

```diff
@@ -2804,6 +2816,55 @@ void SMTLIB2::readAssertNot(LExpr* body)
   _formulas.pushBack(fu);
 }
 
+void SMTLIB2::readAssertSynth(LExpr* forall, LExpr* exist, LExpr* body)
+{
+  pushLookup();
+
+  if (env.options->questionAnswering() != Options::QuestionAnsweringMode::SYNTHESIS) {
+    std::cout << "% WARNING: Found an assert-synth command but synthesis is not enabled. Consider running with '-qa synthesis'." << endl;
+  }
+
+  auto fvars = VList::empty();
+  auto fsorts = SList::empty();
+  auto fRdr = READER(forall);
+  while (fRdr.hasNext()) {
+    auto pRdr = READER(fRdr.readList());
+    auto name = pRdr.readAtom();
+    auto var = TermList::var(_nextVar++);
+    auto sort = parseSort(pRdr.readExpr());
+    tryInsertIntoCurrentLookup(name, var, sort);
+    VList::push(var.var(), fvars);
+    SList::push(sort, fsorts);
+  }
+
+  auto evars = VList::empty();
+  auto esorts = SList::empty();
+  auto eRdr = READER(exist);
+  while (eRdr.hasNext()) {
```

Comment:
Would it make sense to move this loop into a helper function to avoid code duplication?